src/components/ReactorEditor/index.tsx

```diff
@@ -14,41 +14,41 @@ const transform: ValidReactor = (data: Type) => {
 }`
 
 interface Props {
+  name: string
   json: any | undefined
   onSuccess(data: JsWorkerSuccess): void
 }
 
-export function ReactorEditor({ json, onSuccess }: Props) {
-  const [editor, setEditor] =
-    useState<monaco.editor.IStandaloneCodeEditor | null>(null)
-  const monacoDOM = useRef<HTMLDivElement>(null)
-
+export function ReactorEditor({ name, json, onSuccess }: Props) {
   useUpdateMonacoTsTypes(json)
 
-  useEffect(() => {
-    if (monacoDOM) {
-      setEditor((editor) => {
-        if (editor) return editor
-
-        return monaco.editor.create(monacoDOM.current!, {
-          value: DEFAULT_CODE,
-          language: 'typescript',
-        })
-      })
-    }
-
-    return () => editor?.dispose()
-  }, [monacoDOM.current])
-
   const { jsError } = useProessJson({
+    editorName: name,
```

Comment:
Since react can render any arbitrary component whenever they need, monaco editor would be disposed and recreated for new DOM. Handling this manually is very tidious and dangerous. 

Therefore I changed the paradigm into "access by name". I think it's better because someday I'd like to support multi source code, which requires model idetification. Since monaco editor itself is not reactive library, I have to directly access rather reactify it.